### ISSUE TITLE
Cannot consume within typescript project

### ISSUE THREAD

**Issue by Foo42** [testcontainers/testcontainers-node#77]
After installing `testcontainers` my typescript project no longer compiles. This is because `testcontainers`s types depend on `@types/dockerode`, however these are only a dev dependency of `testcontainers` and are thus not installed for consumers.

The fix is to convert `@types/dockerode` into a regular dependency.

See [this thread](https://github.com/Microsoft/types-publisher/issues/81) for further explanation

**Comment by cristianrgreco** [comment]
Hi @Foo42, thanks for raising this. I didn't think testcontainers exposed any Dockerode internals, could you please share an example showing the problem?

**Comment by Foo42** [comment]
I've created a small project to demonstrate the issue https://github.com/Foo42/testcontainers-issue77-demo

Compiling produces the following error:
```
> tsc

node_modules/testcontainers/dist/container.d.ts:2:23 - error TS7016: Could not find a declaration file for module 'dockerode'. '/private/tmp/testcontainers-issue77-demo/node_modules/dockerode/lib/docker.js' implicitly has an 'any' type.
  Try `npm install @types/dockerode` if it exists or add a new declaration (.d.ts) file containing `declare module 'dockerode';`

2 import dockerode from "dockerode";
                        ~~~~~~~~~~~

node_modules/testcontainers/dist/docker-client.d.ts:1:23 - error TS7016: Could not find a declaration file for module 'dockerode'. '/private/tmp/testcontainers-issue77-demo/node_modules/dockerode/lib/docker.js' implicitly has an 'any' type.
  Try `npm install @types/dockerode` if it exists or add a new declaration (.d.ts) file containing `declare module 'dockerode';`

1 import Dockerode from "dockerode";
                        ~~~~~~~~~~~
```
The issue appears to be caused by the fact the `docker-client.d.ts` file exports a class `DockerodeClient` whose constructor had a parameter of type `Dockerode`. Evaluating this file's types therefore requires the type of `Dockerode` be available. These types are not available for consumers of `testcontainers` since they are only a listed as a dev dependency and thus not installed.

It's not something which had occurred to me before hitting this to be honest. I've always instinctively installed `@types` packages as dev dependencies, however I've now realised there are some scenarios where that's insufficient.

**Comment by cristianrgreco** [comment]
Hi @Foo42, thank you that's very helpful. I've merged your PR and deployed it as version 2.8.2, please give it a go and if you're happy then please close the issue 🙂

**Comment by Foo42** [comment]
Sorry took a while getting round to checking, but it seems fixed now (tested with 2.9.0). Thanks
